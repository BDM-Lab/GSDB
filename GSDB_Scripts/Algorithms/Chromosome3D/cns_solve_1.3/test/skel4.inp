REMARKS  skel4.inp
remarks  test skeletonization 
topology @CNS_TOPPAR:protein.top end
parameter @CNS_TOPPAR:protein.param end
segment
  name=tryp
  chain
     @CNS_TOPPAR:protein.link
     sequence
        TRP
     end
  end
end
coor
ATOM      1  HT1 TRP     1       0.101   0.574  -1.385  1.00  0.00      TRYP
ATOM      2  HT2 TRP     1       1.711   0.436  -1.904  1.00  0.00      TRYP
ATOM      3  N   TRP     1       1.067   0.302  -1.101  1.00  0.00      TRYP
ATOM      4  HT3 TRP     1       1.080  -0.689  -0.791  1.00  0.00      TRYP
ATOM      5  CA  TRP     1       1.472   1.192   0.027  1.00  0.00      TRYP
ATOM      6  CB  TRP     1       2.696   2.026  -0.348  1.00  0.00      TRYP
ATOM      7  CG  TRP     1       3.922   1.169  -0.679  1.00  0.00      TRYP
ATOM      8  CD2 TRP     1       4.295   0.679  -1.890  1.00  0.00      TRYP
ATOM      9  CE2 TRP     1       5.424  -0.010  -1.601  1.00  0.00      TRYP
ATOM     10  CE3 TRP     1       3.769   0.774  -3.155  1.00  0.00      TRYP
ATOM     11  CD1 TRP     1       4.813   0.775   0.290  1.00  0.00      TRYP
ATOM     12  NE1 TRP     1       5.727   0.046  -0.320  1.00  0.00      TRYP
ATOM     13  HE1 TRP     1       6.094  -0.743   0.127  1.00  0.00      TRYP
ATOM     14  CZ2 TRP     1       6.138  -0.677  -2.563  1.00  0.00      TRYP
ATOM     15  CZ3 TRP     1       4.472   0.108  -4.148  1.00  0.00      TRYP
ATOM     16  CH2 TRP     1       5.636  -0.605  -3.856  1.00  0.00      TRYP
ATOM     17  C   TRP     1       0.280   2.113   0.309  1.00  0.00      TRYP
ATOM     18  O   TRP     1      -0.402   2.481  -0.636  1.00  0.00      TRYP
ATOM     19  OXT TRP     1      -0.106   2.192   1.459  1.00  0.00      TRYP
END
do (b=20) ( all )
{===================================================================}
{* test with arbitrary unit cell, P1 space group *}
{===================================================================}
xray
  a=16.2 b=16.5 c=20. alpha=90.  beta=90.  gamma=90.

   declare name=fobs type=complex domain=reciprocal end
   declare name=fpart domain=reciprocal type=complex end
   declare name=fcalc domain=reciprocal type=complex end
   declare name=fom domain=reciprocal type=real end
   declare name=weight domain=reciprocal type=real end
   declare name=sigma domain=reciprocal type=real end
   declare name=test domain=reciprocal type=integer end
  eval ($SG="P1" )
  @CNS_XTALLIB:spacegroup.lib  (SG=$SG )
   SCATter ( chemical C* ) 
      2.31000 20.8439 1.02000 10.2075 1.58860 .568700 .865000 51.6512 .215600
   SCATter ( chemical N* )
      12.2126 .005700 3.13220 9.89330 2.01250 28.9975 1.16630 .582600 -11.529
   SCATter ( chemical O* )
      3.04850 13.2771 2.28680 5.70110 1.54630 .323900 .867000 32.9089 .250800
   SCATter ( chemical S* )
      6.90530 1.46790 5.20340 22.2151 1.43790 .253600 1.58630 56.1720 .866900
   SCATter ( chemical P* )
      6.43450 1.90670 4.17910 27.1570 1.78000 0.52600 1.49080 68.1645 1.11490
   SCATter ( chemical FE* )
      11.1764 4.61470 7.38630 0.30050 3.39480 11.6729 0.07240 38.5566 0.97070
  method=fft 
  evaluate ($lr=10.)
  evaluate ($hr=2.) 
  binresolution $lr $hr 
  mapresolution $hr
  tselection=( $hr <= d <= $lr  ) 
  nreflections=10000
  generate  10. 2. 
  fft 
    grid=0.25 
  end
  declare name=e_map domain=real end
  associate fcalc ( resid 1:20 ) 
  predict mode=real
     to=e_map 
     atomselection=( not hydro ) 
  end
  declare name=before_map domain=real end
  do ( before_map=e_map ) ( all )
  set seed 666 end
  do ( FCALC =FT(e_map) + gauss(25.0) ) ( all )
  do ( e_map =FT(FCALC) ) ( all )
  show sum (mult e_map before_map) ( all)
  evaluate ($numerator=$result) 
  show sum (mult  e_map e_map) ( all )
  evaluate ($denominator=$result)
  show sum (mult  before_map before_map) ( all )
  evaluate ($denominator=sqrt( $denominator * $result )) 
  evaluate ($corr=$numerator/ $denominator) 
  display  $corr   (before noise added - after noise added)
  show ave (e_map) (all)
  evaluate ($e_map_ave = $RESULT)
  show ave (e_map^2) (all)
  evaluate ($sigma = sqrt ( $RESULT - $e_map_ave^2))
  do (e_map=e_map/$sigma) ( all ) 
  skel
    REMARKS minout=1.0
    from=e_maP
    maxden=4.0
    minden=1.8
    epden=3.0
    MING=21
    MXNF=2
    Peakden=2.0
    MXSD = 10000
    selection=( all )
    B=5.
  end
  show sum (mult e_map before_map) ( all)
  evaluate ($numerator=$result) 
  show sum (mult  e_map e_map) ( all )
  evaluate ($denominator=$result)
  show sum (mult  before_map before_map) ( all )
  evaluate ($denominator=sqrt( $denominator * $result )) 
  evaluate ($corr=$numerator/ $denominator) 
  display  $corr   (before noise added - after noise added, after skeleton.)
end
stop
