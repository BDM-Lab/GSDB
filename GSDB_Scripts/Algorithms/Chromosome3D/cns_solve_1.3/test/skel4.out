          ============================================================
          |                                                          |
          |            Crystallography & NMR System (CNS)            |
          |                         CNSsolve                         |
          |                                                          |
          ============================================================
           Version: 1.3
           Status: General release
          ============================================================
           Written by: A.T.Brunger, P.D.Adams, G.M.Clore, W.L.DeLano,
                       P.Gros, R.W.Grosse-Kunstleve,J.-S.Jiang,J.M.Krahn,
                       J.Kuszewski, M.Nilges, N.S.Pannu, R.J.Read,
                       L.M.Rice, G.F.Schroeder, T.Simonson, G.L.Warren.
           Copyright (c) 1997-2010 Yale University
          ============================================================
           Running on machine: lewis4-r710-login-node223 (x86_64/Linux,64-bit)
                               with 16 threads
           Program started by: oeow39
           Program started at: 16:08:07 on 05-Jul-2018
          ============================================================

 FFT3C: Using FFTPACK4.1

 CNSsolve>REMARKS  skel4.inp 
 CNSsolve>remarks  test skeletonization 
 CNSsolve>topology @CNS_TOPPAR:protein.top end 
 ASSFIL: file protein.top opened.
 RTFRDR>remarks file toppar/protein.top 
 RTFRDR>remarks   protein topology with polar hydrogens 
 RTFRDR> 
 RTFRDR>! 
 RTFRDR>! Please cite the following reference when using these parameters: 
 RTFRDR>! Engh, R.A. and  Huber, R. (1991). Accurate Bond and 
 RTFRDR>!  Angle Parameters for X-ray Protein-Structure Refinement, 
 RTFRDR>!  Acta Cryst. A47, 392-400. 
 RTFRDR>! 
 RTFRDR>! 
 RTFRDR> 
 RTFRDR>set message ? end eval ($old_message=$result) set echo ? end eval ($old_echo=$result) 
 MESSage=NORM
 EVALUATE: symbol $OLD_MESSAGE set to "NORM" (string)
 ECHO=TRUE {ON}
 EVALUATE: symbol $OLD_ECHO set to TRUE (logical)
 RTFRDR>set message=off echo=off end 
 RTFRDR> 
 RTFRDR> 
 RTFRDR> end 
 CNSsolve>parameter @CNS_TOPPAR:protein.param end 
 ASSFIL: file protein.param opened.
 PARRDR>remarks file toppar/protein.param 
 PARRDR>remarks   protein parameters with Lennard-Jones van der Waals parameters 
 PARRDR>remarks   for crystallographic structure determination 
 PARRDR> 
 PARRDR>! 
 PARRDR>! Please cite the following reference when using these parameters: 
 PARRDR>! Engh, R.A. and  Huber, R. (1991). Accurate Bond and 
 PARRDR>!  Angle Parameters for X-ray Protein-Structure Refinement, 
 PARRDR>!  Acta Cryst. A47, 392-400. 
 PARRDR>! 
 PARRDR>! 
 PARRDR> 
 PARRDR>set message ? end eval ($old_message=$result) set echo ? end eval ($old_echo=$result) 
 MESSage=NORM
 EVALUATE: symbol $OLD_MESSAGE set to "NORM" (string)
 ECHO=TRUE {ON}
 EVALUATE: symbol $OLD_ECHO set to TRUE (logical)
 PARRDR>set message=off echo=off end 
 PARRDR> 
 PARRDR> end 
 CNSsolve>segment 
 SEGMENT>  name=tryp 
 SEGMENT>  chain 
 CHAIN>     @CNS_TOPPAR:protein.link 
 ASSFIL: file protein.link opened.
 CHAIN>remarks file toppar/protein.link 
 CHAIN>remarks 
 CHAIN>remarks this is a macro to define standard protein peptide bonds 
 CHAIN>remarks and termini to generate a protein sequence. 
 CHAIN> 
 CHAIN>set message ? end eval ($old_message=$result) set echo ? end eval ($old_echo=$result) 
 MESSage=NORM
 EVALUATE: symbol $OLD_MESSAGE set to "NORM" (string)
 ECHO=TRUE {ON}
 EVALUATE: symbol $OLD_ECHO set to TRUE (logical)
 CHAIN>set message=off echo=off end 
 CHAIN>     sequence 
 SEQUENCE>        TRP 
 SEQUence-element (terminate with END) =     end 
 SEGMNT:     1 residues were inserted into segment "TRYP"
 CHAIN>  end 
 SEGMENT>end 
 Status of internal molecular topology database:
 -> NATOM=         19(MAXA=     1000000)  NBOND=         20(MAXB=     1000000)
 -> NTHETA=        30(MAXT=     2000000)  NGRP=           3(MAXGRP=   1000000)
 -> NPHI=          10(MAXP=     2000000)  NIMPHI=        15(MAXIMP=   1000000)
 -> NNB=            0(MAXNB=    1000000) 
 CNSsolve>coor 
 COOR>ATOM      1  HT1 TRP     1       0.101   0.574  -1.385  1.00  0.00      TRYP 
 COOR>ATOM      2  HT2 TRP     1       1.711   0.436  -1.904  1.00  0.00      TRYP 
 CNSsolve>do (b=20) ( all ) 
 SELRPN:     19 atoms have been selected out of     19
 CNSsolve>{===================================================================} 
 CNSsolve>{* test with arbitrary unit cell, P1 space group *} 
 CNSsolve>{===================================================================} 
 CNSsolve>xray 
 XRAY>  a=16.2 b=16.5 c=20. alpha=90.  beta=90.  gamma=90. 
 XRAY> 
 XRAY>   declare name=fobs type=complex domain=reciprocal end 
 XDECLARE: Object FOBS has been declared.
 XRAY>   declare name=fpart domain=reciprocal type=complex end 
 XDECLARE: Object FPART has been declared.
 XRAY>   declare name=fcalc domain=reciprocal type=complex end 
 XDECLARE: Object FCALC has been declared.
 XRAY>   declare name=fom domain=reciprocal type=real end 
 XDECLARE: Object FOM has been declared.
 XRAY>   declare name=weight domain=reciprocal type=real end 
 XDECLARE: Object WEIGHT has been declared.
 XRAY>   declare name=sigma domain=reciprocal type=real end 
 XDECLARE: Object SIGMA has been declared.
 XRAY>   declare name=test domain=reciprocal type=integer end 
 XDECLARE: Object TEST has been declared.
 XRAY>  eval ($SG="P1" ) 
 EVALUATE: symbol $SG set to "P1" (string)
 XRAY>  @CNS_XTALLIB:spacegroup.lib  (SG=$SG ) 
 ASSFIL: file spacegroup.lib opened.
 XRAY>! file  libraries/xtal/spacegroup.lib 
 XRAY>! library of symmetry operators for CNS 
 XRAY>! Automatically generated with SgInfo Version 2.00a13 
 XRAY> 
 XRAY>module 
 MODULE-DECLARATION>( 
 MODULE-DECLARATION>  &sg=$sg; 
 MODULE-DECLARATION>  &sgparam=$sgparam; 
 MODULE-DECLARATION>) 
 &_2_SG set to $_1_SG 
 XRAY> 
 XRAY>set message ? end 
 MESSage=NORM
 XRAY>evaluate ($message_old=$result) 
 EVALUATE: symbol $MESSAGE_OLD set to "NORM" (string)
 XRAY>set echo ? end 
 ECHO=TRUE {ON}
 XRAY>evaluate ($echo_old=$result) 
 EVALUATE: symbol $ECHO_OLD set to TRUE (logical)
 XRAY>set echo=off message=off end 
SYMMETRY: found symmetry operators for space group P1 in library
 XRAY> 
 XRAY>   SCATter ( chemical C* ) 
 SELRPN:     11 atoms have been selected out of     19
 SCATTER_A1=      2.31000 20.8439 1.02000 10.2075 1.58860 .568700 .865000 51.6512 .215600 
 XRAY>   SCATter ( chemical N* ) 
 SELRPN:      2 atoms have been selected out of     19
 SCATTER_A1=      12.2126 .005700 3.13220 9.89330 2.01250 28.9975 1.16630 .582600 -11.529 
 XRAY>   SCATter ( chemical O* ) 
 SELRPN:      2 atoms have been selected out of     19
 SCATTER_A1=      3.04850 13.2771 2.28680 5.70110 1.54630 .323900 .867000 32.9089 .250800 
 XRAY>   SCATter ( chemical S* ) 
 SELRPN:      0 atoms have been selected out of     19
 SCATTER_A1=      6.90530 1.46790 5.20340 22.2151 1.43790 .253600 1.58630 56.1720 .866900 
 XRAY>   SCATter ( chemical P* ) 
 SELRPN:      0 atoms have been selected out of     19
 SCATTER_A1=      6.43450 1.90670 4.17910 27.1570 1.78000 0.52600 1.49080 68.1645 1.11490 
 XRAY>   SCATter ( chemical FE* ) 
 SELRPN:      0 atoms have been selected out of     19
 SCATTER_A1=      11.1764 4.61470 7.38630 0.30050 3.39480 11.6729 0.07240 38.5566 0.97070 
 XRAY>  method=fft 
 XRAY>  evaluate ($lr=10.) 
 EVALUATE: symbol $LR set to    10.0000     (real)
 XRAY>  evaluate ($hr=2.) 
 EVALUATE: symbol $HR set to    2.00000     (real)
 XRAY>  binresolution $lr $hr 
 XRAY>  mapresolution $hr 
 XRAY>  tselection=( $hr <= d <= $lr  ) 
 XRAY>  nreflections=10000 
 XRAY: allocating space for   10000 reflections.
 XRAY>  generate  10. 2. 
 XGENER: generating reflections to produce a full set
         for the specified resolution range.
 XGENE2:     1373 new reflections have been generated.
 XRAY>  fft 
 XFFT>    grid=0.25 
 XFFT>  end 
 XRAY>  declare name=e_map domain=real end 
 XRAY>  associate fcalc ( resid 1:20 ) 
 SELRPN:     19 atoms have been selected out of     19
 XRAY>  predict mode=real 
 PREDict>     to=e_map 
 PREDict>     atomselection=( not hydro ) 
 SELRPN:     15 atoms have been selected out of     19
 PREDict>  end 
 FCALC: #scatt.=     15 #anomalous=   0 #special pos.=  0 occupancies=1
 XMAPASU: using grid [  32,  36,  40] and sublattice [  32,  36,  40]
 Minimum brick that covers asymmetric unit:
   A=     0,...,    32  B=     0,...,    36  C=     0,...,    40
 XMAPAL: allocating space for real space object.
 XRAY>  declare name=before_map domain=real end 
 XRAY>  do ( before_map=e_map ) ( all ) 
 XMAPAL: allocating space for real space object.
 Total of     46080 map elements were selected.
 XRAY>  set seed 666 end 
 XRAY>  do ( FCALC =FT(e_map) + gauss(25.0) ) ( all ) 
 XMAPAL: allocating space for real space object.
 Total of     46080 map elements were selected.
 XSFAL: allocating space for complex reciprocal space object.
 XSFAL: allocating space for complex reciprocal space object.
 Total of      1373 structure factor elements were selected.
 XRAY>  do ( e_map =FT(FCALC) ) ( all ) 
 XSFAL: allocating space for complex reciprocal space object.
 Total of      1373 structure factor elements were selected.
 XRAY>  show sum (mult e_map before_map) ( all) 
 Sum of     46080 elements =               556.1546
 XRAY>  evaluate ($numerator=$result) 
 EVALUATE: symbol $NUMERATOR set to    556.155     (real)
 XRAY>  show sum (mult  e_map e_map) ( all ) 
 Sum of     46080 elements =              3346.2798
 XRAY>  evaluate ($denominator=$result) 
 EVALUATE: symbol $DENOMINATOR set to    3346.28     (real)
 XRAY>  show sum (mult  before_map before_map) ( all ) 
 Sum of     46080 elements =               798.4195
 XRAY>  evaluate ($denominator=sqrt( $denominator * $result )) 
 EVALUATE: symbol $DENOMINATOR set to    1634.54     (real)
 XRAY>  evaluate ($corr=$numerator/ $denominator) 
 EVALUATE: symbol $CORR set to   0.340251     (real)
 XRAY>  display  $corr   (before noise added - after noise added) 
 0.340251   (before noise added - after noise added)
 XRAY>  show ave (e_map) (all) 
 SHOW: average of     46080 elements=                 0.0000
 XRAY>  evaluate ($e_map_ave = $RESULT) 
 EVALUATE: symbol $E_MAP_AVE set to   0.102703E-10 (real)
 XRAY>  show ave (e_map^2) (all) 
 SHOW: average of     46080 elements=                 0.0726
 XRAY>  evaluate ($sigma = sqrt ( $RESULT - $e_map_ave^2)) 
 EVALUATE: symbol $SIGMA set to   0.269479     (real)
 XRAY>  do (e_map=e_map/$sigma) ( all ) 
 Total of     46080 map elements were selected.
 XRAY>  skel 
 SKELeton>    REMARKS minout=1.0 
 SKELeton>    from=e_maP 
 SKELeton>    maxden=4.0 
 SKELeton>    minden=1.8 
 SKELeton>    epden=3.0 
 SKELeton>    MING=21 
 SKELeton>    MXNF=2 
 SKELeton>    Peakden=2.0 
 SKELeton>    MXSD = 10000 
 SKELeton>    selection=( all ) 
 XMPSELE: total of    46080 map elements were selected.
 SKELeton>    B=5. 
 SKELeton>  end 
 |----------------------- SKELeton ------------------------------------|
 | B =5.00   EPDEn= 3.00   MINGraph=  21   MAxGraphStats =  500        |
 | MAXDen=  4.00  MINDen=  1.80  PEAKden=  2.00 MaXSearchDepth=  10000 |
 | FROM = E_MAP                 MaXNodeFrac=   2   SMallED=0.00050     |
 |---------------------------------------------------------------------|
  1524 nodes were identified.
  1233 nodes were removed during thinning.
   291 nodes remain after thinning.
    14 isolated nodes were pruned.
    11 chains with   2 nodes each were pruned.
     4 chains with   3 nodes each were pruned.
     3 chains with   4 nodes each were pruned.
     4 chains with   5 nodes each were pruned.
     1 chain  with   6 nodes was pruned.
     1 chain  with   7 nodes was pruned.
     1 chain  with   8 nodes was pruned.
     1 chain  with   9 nodes was pruned.
     1 chain  with  10 nodes was pruned.
     1 chain  with  13 nodes was pruned.
     1 chain  with 158 nodes remains.
A total of   133 nodes were removed during pruning.
The largest chain has   158 nodes,100.0% of the   158 remaining nodes
 XRAY>  show sum (mult e_map before_map) ( all) 
 Sum of     46080 elements =               898.3051
 XRAY>  evaluate ($numerator=$result) 
 EVALUATE: symbol $NUMERATOR set to    898.305     (real)
 XRAY>  show sum (mult  e_map e_map) ( all ) 
 Sum of     46080 elements =              1282.6949
 XRAY>  evaluate ($denominator=$result) 
 EVALUATE: symbol $DENOMINATOR set to    1282.69     (real)
 XRAY>  show sum (mult  before_map before_map) ( all ) 
 Sum of     46080 elements =               798.4195
 XRAY>  evaluate ($denominator=sqrt( $denominator * $result )) 
 EVALUATE: symbol $DENOMINATOR set to    1011.99     (real)
 XRAY>  evaluate ($corr=$numerator/ $denominator) 
 EVALUATE: symbol $CORR set to   0.887660     (real)
 XRAY>  display  $corr   (before noise added - after noise added, after skeleton.) 
 0.88766   (before noise added - after noise added, after skeleton.)
 XRAY>end 
 CNSsolve>stop 
          ============================================================
           Maximum dynamic memory allocation:    11485840 bytes
           Maximum dynamic memory overhead:           720 bytes
           Program started at: 16:08:07 on 05-Jul-2018
           Program stopped at: 16:08:07 on 05-Jul-2018
           CPU time used:       1.0538 seconds
          ============================================================
